src/engine/math_util.c

```diff
@@ -541,47 +617,76 @@ void mtxf_lookat(Mat4 mtx, Vec3f from, Vec3f to, s32 roll) {
  * 'mtx' is the look-at matrix from the camera.
  * 'position' is the position of the object in the world.
  * 'scale' is the scale of the object.
- * 'angle' rotates the object while still facing the camera.
+ * 'roll' rotates the object while still facing the camera.
  */
-void mtxf_billboard(Mat4 dest, Mat4 mtx, Vec3f position, Vec3f scale, s32 angle) {
-    register s32 i;
-    register f32 sx = scale[0];
-    register f32 sy = scale[1];
-    register f32 sz = ((f32 *) scale)[2];
-    register f32 *temp2, *temp = (f32 *)dest;
-    for (i = 0; i < 16; i++) {
-        *temp = 0;
-        temp++;
+void mtxf_billboard(Mat4 dest, Mat4 src, Vec3f position, Vec3f scale, s32 roll) {
+    f32 sx = scale[0];
+    f32 sy = scale[1];
+    f32 sz = scale[2];
+    f32 *dstp = (f32 *)dest;
+    s32 i;
+    for (i = 0; i < 12; i++) {
+        *dstp++ = 0;
```

Comment:
breathing room before and after new blocks that have normal statements before or after
```suggestion
    s32 i;

    for (i = 0; i < 12; i++) {
        *dstp++ = 0;
```